docs/index.bs

```diff
@@ -3359,6 +3348,31 @@ spec: rfc7231; urlPrefix: https://tools.ietf.org/html/rfc7231
     </ol>
   </section>
 
+  <section algorithm="extend-service-worker-lifetime-algorithm">
+    <h3 id="extend-service-worker-lifetime-algorithm" dfn>Extend Service Worker Lifetime</h3>
+
+    <dl>
+      <dt>Input</dt>
+        <dd><var>event</var>, an {{ExtendableEvent}} object</dd>
+      <dt>Output</dt>
+        <dd>None</dd>
+    </dl>
+    <ol>
+      <li>If <var>event</var>'s <a>extend lifetime promises</a> is empty, unset <var>event</var>'s <a>extensions allowed flag</a> and abort these steps.</li>
+      <li>Let <var>extendLifetimePromises</var> be an empty array.</li>
+      <li>Run the following substeps <a>in parallel</a>:
+        <ol>
+          <li><em>SetupPromiseArray</em>: Set <var>extendLifetimePromises</var> to a copy of <var>event</var>'s <a>extend lifetime promises</a>.</li>
```

Comment:
I thought we had a better way of doing this? Just letting the promises decrease a counter.